AzureMonitor: Ensure original query properties are included unchanged

aangelisc opened the pull request:
<!--

Thank you for sending a pull request! Here are some tips:

1. If this is your first time, please read our contribution guide at https://github.com/grafana/grafana/blob/main/CONTRIBUTING.md

2. Ensure you include and run the appropriate tests as part of your Pull Request.

3. In a new feature or configuration option, an update to the documentation is necessary. Everything related to the documentation is under the docs folder in the root of the repository.

4. If the Pull Request is a work in progress, make use of GitHub's "Draft PR" feature and mark it as such.

5. If you can not merge your Pull Request due to a merge conflict, Rebase it. This gets it in sync with the main branch.

6. Name your PR as "<FeatureArea>: Describe your change", e.g. Alerting: Prevent race condition. If it's a fix or feature relevant for the changelog describe the user impact in the title. The PR title is used to auto-generate the changelog for issues marked with the "add to changelog" label.

-->

Fixes #49641

aangelisc committed:
Ensure original query properties are included unchanged

andresmgot commented on public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_log_analytics/azure_log_analytics_datasource.ts:
```diff
@@ -134,6 +134,7 @@ export default class AzureLogAnalyticsDatasource extends DataSourceWithBackend<
 
     return {
+      ...target,
       refId: target.refId,
```
you can know remove this line (the same for others)

aangelisc replied:
I may delete this PR in favour of #49543

andresmgot replied:
I still think that having both contributes to more robust code :)

andresmgot reviewed (commented):
Can you add tests for this? Thanks!

aangelisc committed:
Remove refId from pseudo datasources

aangelisc committed:
Include testing for interpolateVariablesInQueries

- Add util function to create template variables
- Update mock query with missing props
- Additional tests on each ds for template variables

andresmgot commented on public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_log_analytics/azure_log_analytics_datasource.test.ts:
```diff
@@ -369,4 +372,37 @@ describe('AzureLogAnalyticsDatasource', () => {
+      const query = createMockQuery();
+      query.queryType = AzureQueryType.LogAnalytics;
+      const templatedQuery = ctx.ds.interpolateVariablesInQueries([query], {});
+      expect(templatedQuery[0]).toEqual(query);
```
we can add a check here to make sure that query.datasource is defined to be extra safe

aangelisc replied:
This is a good shout 😊

andresmgot reviewed (approved):
LGTM!

aangelisc committed:
Correct typo and add explicit check for datasource
